src/templates/pluginVuex.ejs

```diff
@@ -0,0 +1,127 @@
+/* eslint-disable no-await-in-loop */
+/*
+ * Copyright © Bold Brand Commerce Sp. z o.o. All rights reserved.
+ * See LICENSE for license details.
+ */
+import deepmerge from 'deepmerge';
+import extendsModules from '~/.nuxt/extends.modules';
+import storeModules from '~/.nuxt/store.modules';
+
+export default async ({ app }) => {
+    const storesForExtend = {};
+    const asyncForEach = async (array, callback) => {
+        for (let index = 0; index < array.length; index += 1) {
+            await callback(array[index], index, array);
+        }
+    };
+
+    await asyncForEach(Object.values(extendsModules), async (extendContent) => {
+        if (extendContent.extendStore) {
+            await asyncForEach(Object.keys(extendContent.extendStore), async (extendName) => {
+                const tmpStore = {};
+                const storeContent = await extendContent.extendStore[extendName]();
+
+                if (storeContent.state) {
+                    tmpStore.state = storeContent.state();
+                }
+                if (storeContent.actions) {
+                    tmpStore.actions = storeContent.actions;
+                }
+                if (storeContent.mutations) {
+                    tmpStore.mutations = storeContent.mutations;
+                }
+                if (storeContent.getters) {
+                    tmpStore.getters = storeContent.getters;
+                }
+
+                storesForExtend[extendName] = deepmerge(
+                    storesForExtend[extendName] || {},
+                    tmpStore,
+                );
+            });
+        }
+    });
+
+    app.store.registerModule('index', {
+        actions: {
+            resetState({ dispatch }) {
+                Object.keys(storeModules).forEach((storeName) => {
+                    if (app.store._actions[`${storeName}/__clearStorage`]
+                            && app.store._actions[`${storeName}/__clearStorage`].length) {
+                        dispatch(`${storeName}/__clearStorage`);
+                    }
+                });
+            },
+        },
+    });
+
+    Object.keys(storeModules).forEach((storeName) => {
+        const moduleIsRegistered = app.store._modules.root._children[storeName] !== undefined;
+        const stateExists = app.store.state[storeName];
+        let states = storeModules[storeName].state();
+
+        if (!storeModules[storeName].mutations) {
+            storeModules[storeName].mutations = {};
+        }
+        if (!storeModules[storeName].actions) {
+            storeModules[storeName].actions = {};
+        }
+
+        if (storesForExtend[storeName]) {
+            if (storesForExtend[storeName].state) {
```

Comment:
You may Iterate via Object.keys(storesForExtend[storeName]) without checking for each individual